### PR TITLE
Template the ingress controller 

### PR DESCRIPTION
**Pull request by bprashanth**
We still need https://github.com/kubernetes/contrib/pull/791 to run the controller as a static pod
ref https://github.com/kubernetes/kubernetes/issues/23663

### CODE REVIEW

**Comment by bprashanth** [comment]
@roberthbailey @andyzheng0831 PTAL
First commit pipes a timestamp uid as a salt pillar
Second commit templatizes the rc
Last fixes e2es

Soon after this I plan to push a new glbc image with the changes required to run it on the master, and if I get ~a week of green, we can flip the switch to running a static pod

**Comment by andyzheng0831** [comment]
It looks to me that the only work on my side is to handle the new env var kube_uid in the manifest file. The fix should be very straightforward. I will make a PR, but need to wait for this one being merged first, so that I can test my fix.

**Comment by bprashanth** [comment]
Reassigning to Minhan assuming Robbie is busy, @kubernetes/goog-cluster

File: cluster/common.sh [hunk 1]
```diff
@@ -202,6 +202,16 @@ function gen-kube-bearertoken() {
...
 }
 
+# Generate uid
+# This function only works on systems with python. It generates a time based
+# UID instead of a UUID because GCE has a name length limit.
+#
+# Vars set:
+#   KUBE_UID
+function gen-uid {
+    KUBE_UID=$(python -c 'import uuid; print uuid.uuid1().fields[0]')
```
Comment (freehan) [inline cluster/common.sh]: Maybe consider this? Your choice. I am okay with python. 
`cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1`

**Comment by freehan** [comment]
LGTM

File: cluster/common.sh (hunk 1 above)
Comment (bprashanth) [inline cluster/common.sh]: yeah i thought about urandom and in fact my previous commit had /proc/sys/kernel/random/uuid, but neither of those is  cross platform. You could argue that python might not be installed, but I grepped our kube-up scripts and it showed that other places were using python so i figured it's the least evil choice.

Comment (roberthbailey) [inline cluster/common.sh]: I think that @mikedanese is working on removing python for the on-host parts, but maybe it's ok to leave in here.

**Comment by bprashanth** [comment]
@k8s-bot unit test this github issue: #25138

File: cluster/addons/cluster-loadbalancing/glbc/glbc.yaml [hunk 2]
```diff
@@ -1,3 +1,8 @@
+{% set kube_uid = "" -%}
```
Comment (roberthbailey) [inline cluster/addons/cluster-loadbalancing/glbc/glbc.yaml]: @andyzheng0831 This file is getting templated in preparation for running as a static pod on the master. FYI.

Comment (andyzheng0831) [inline cluster/addons/cluster-loadbalancing/glbc/glbc.yaml]: Thanks for heads-up. I am waiting for this PR to be merged, then I can test my change to gci support

**Comment by bprashanth** [comment]
@k8s-bot test this github issue: #25170

**Comment by bprashanth** [comment]
P1 ing this because it's been sitting in the queue for 20 days

**Pull request merged**